### ISSUE TITLE
Auth failure on 2.0.1 (works on 2.0.0)

### ISSUE THREAD

**Issue by calvarez-ov** [ajoberstar/grgit#200]
Hello:
We have this task:
```gradle
task createTag() {
    doLast {
        def tagName = "v${android.defaultConfig.versionName}"
        println "tagging ${tagName}"
        def grgit = Grgit.open(dir: project.rootDir)
        grgit.tag.add {
            name = tagName
            message = "Release of ${tagName}"
        }
        println "pushing to origin"
        grgit.push(tags: true)
        grgit.close()
    }
}
```

If we use plugin version 2.0.0, the tag is pushed fine.
If we use plugin version 2.0.1, we have an error "Auth fail".

Here's the exception:
```
org.eclipse.jgit.api.errors.TransportException: git@github.com:ourorg/our-repo.git: Auth fail

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':library:createTag'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:100)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:62)
	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:97)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:87)
	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:625)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:580)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.internal.UncheckedException: org.eclipse.jgit.api.errors.TransportException: git@github.com:ourorg/our-repo.git: Auth fail
	at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:63)
	at org.gradle.internal.UncheckedException.throwAsUncheckedException(UncheckedException.java:40)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:151)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:110)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
	... 28 more
Caused by: org.eclipse.jgit.api.errors.TransportException: git@github.com:ourorg/our-repo.git: Auth fail
	at org.eclipse.jgit.api.PushCommand.call(PushCommand.java:180)
	at org.eclipse.jgit.api.PushCommand.call(PushCommand.java:85)
	at java_util_concurrent_Callable$call$0.call(Unknown Source)
	at org.ajoberstar.grgit.operation.PushOp.call(PushOp.groovy:85)
	at org.ajoberstar.grgit.operation.PushOp.call(PushOp.groovy)
	at java_util_concurrent_Callable$call.call(Unknown Source)
	at java_util_concurrent_Callable$call.call(Unknown Source)
	at org.ajoberstar.grgit.internal.OpSyntax.mapOperation(OpSyntax.groovy:33)
	at org.ajoberstar.grgit.internal.OpSyntax$mapOperation.callStatic(Unknown Source)
	at org.ajoberstar.grgit.Grgit.push(Grgit.groovy)
	at org.ajoberstar.grgit.Grgit$push$0.call(Unknown Source)
	at publish_dnb6rv0z06v6nxm68uuyi5lvz$_run_closure7$_closure28.doCall(/jenkins/workspace/ci/android-team/test-job/library/publish.gradle:152)
	at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:700)
	at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:673)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:121)
	... 34 more
Caused by: org.eclipse.jgit.errors.TransportException: git@github.com:ourorg/our-repo.git: Auth fail
	at org.eclipse.jgit.transport.JschConfigSessionFactory.getSession(JschConfigSessionFactory.java:163)
	at org.eclipse.jgit.transport.SshTransport.getSession(SshTransport.java:137)
	at org.eclipse.jgit.transport.TransportGitSsh$SshPushConnection.<init>(TransportGitSsh.java:332)
	at org.eclipse.jgit.transport.TransportGitSsh.openPush(TransportGitSsh.java:174)
	at org.eclipse.jgit.transport.PushProcess.execute(PushProcess.java:160)
	at org.eclipse.jgit.transport.Transport.push(Transport.java:1310)
	at org.eclipse.jgit.api.PushCommand.call(PushCommand.java:169)
	... 48 more
Caused by: com.jcraft.jsch.JSchException: Auth fail
	at com.jcraft.jsch.Session.connect(Session.java:519)
	at org.eclipse.jgit.transport.JschConfigSessionFactory.getSession(JschConfigSessionFactory.java:117)
	... 54 more
```

**Comment by calvarez-ov** [comment]
Note: We managed to use version 2.0.1 successfully as described above, a month ago.  Then we updated multiple jenkins plugins.  Not sure if one of them could have introduced an issue.

We have more jenkins plugins to update now.  I'll post an update here after updating them and testing again.

**Comment by calvarez-ov** [comment]
Updating to the latest version of all plugins didn't resolve the issue.

For info, here's the plugin update which caused the issue to appear:
![image](https://user-images.githubusercontent.com/9365138/31680254-167e599c-b374-11e7-9286-69f32a8712ea.png)

**Comment by ajoberstar** [comment]
Does your current Jenkins config work with 2.0.0? Looking at the diff from [2.0.0..2.0.1](https://github.com/ajoberstar/grgit/compare/2.0.0...2.0.1), I don't think anything could have introduced an auth fail.

Which [authentication approach](http://ajoberstar.org/grgit/grgit-authentication.html) are you trying to use?

**Comment by calvarez-ov** [comment]
We're using ssh-agent.

I also looked at the 2.0.0-2.0.1 diff and didn't notice much, except possibly the version of jgit. What's strange is that in that diff, you don't see a bump in the jgit version.  This bump is included in the release notes for grgit 2.1.0, but when looking at the gradle logs, I saw that jgit 4.9.0 was already being used with grgit 2.0.1.  I actually tried to force this back down to 4.8.0 to see if that could be the problem, but I didn't succeed in forcing the jgit version.

grgit 2.0.0 worked fine after the jenkins upgrade in this screenshot. I haven't tried 2.0.0 again since we updated jenkins again. I'll try it (and I guess grgit 2.1.0 too) and report back here.

**Comment by calvarez-ov** [comment]
On our current jenkins configuration:
* We can push tags with grgit 2.0.0
* We get "auth fail" errors trying to push tags with grgit 2.0.1 and 2.1.0

The list of all the Jenkins plugins is too big to put here. Here are some that might be relevant (just looking at anything related to authentication or git). Let me know if you need more info:
* Authentication Tokens API plugin 1.3
* Credentials Binding Plugin 1.13
* Credentials Plugin 2.1.16
* Git client plugin 2.5.0
* Git plugin 3.6.0
* Jsch dependency plugin 0.1.54.1
* SSH Agent plugin 1.15
* SSH Credentials plugin 1.13

These are all at the latest version.

**Comment by ajoberstar** [comment]
Looking back at the POM for 2.0.1 I see that it has ranges in it (rather than explicit versions like 2.0.0). That would coincide with my switch to using dependency-lock, which I thought would be reflected in my POM file, but it seems that was a misunderstanding on my part.

That leans me towards JGit being the change most likely to cause an issue. Here are [the 4.9 release notes]. I won't have a chance to look into this more tonight, but should later this week. Let me know if you find anything more in the meantime.

**Comment by ajoberstar** [comment]
I wonder if it could be tied at all to this [issue](https://bugs.eclipse.org/bugs/show_bug.cgi?id=490939) resolved in 4.9. Release notes describe it as:

> Let Jsch know about ~/.ssh/config. Ensure the Jsch instance used knows about ~/.ssh/config. This enables Jsch to honor more user configurations, in particular also the UserKnownHostsFile configuration, or additional identities given via multiple IdentityFile entries.

If your Jenkins has a config file like that, maybe something changes the behavior from what happened with 4.8.

**Comment by ajoberstar** [comment]
Closing due to lack of response. Feel free to reopen, if you want to continue discussing this.
